source/Mlos.Python/mlos/Spaces/HypergridAdapters/CategoricalToOneHotEncodedHypergridAdapter.py

```diff
@@ -0,0 +1,260 @@
+from mlos.Spaces import CategoricalDimension, DiscreteDimension, Hypergrid, SimpleHypergrid
+from mlos.Spaces.HypergridAdapters.HypergridAdapter import HypergridAdapter
+from mlos.Spaces.HypergridAdapters.HierarchicalToFlatHypergridAdapter import HierarchicalToFlatHypergridAdapter
+from mlos.Spaces.HypergridAdapters.CategoricalToDiscreteHypergridAdapter import CategoricalToDiscreteHypergridAdapter
+
+
+class CategoricalToOneHotEncodingAdapteeTargetMapping:
+    def __init__(self, one_hot_encoder: OneHotEncoder):
+        self.target_dims = []
+        self.one_hot_encoder = one_hot_encoder
+        self.num_dummy_dims = 0
+
+
+class CategoricalToOneHotEncodedHypergridAdapter(HypergridAdapter):
+    """ Maps values in categorical dimensions into values in OneHotEncoded dimensions using:
+        https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html,
+        which will be referred to as sklearn's OHE.
+
+        Parameters
+        ----------
+        merge_all_categorical_dimensions: bool
+            If True, sklearn's OHE will be applied to the cross product of all categorical levels in the adaptee space.
+            If False, sklearn's OHE will be applied individually to each categorical dimension in the adaptee space.
+            Default=False.
+        drop: {None, 'first', 'if_binary'}
+            Argument passed to sklearn's OHE with same argument name.
+            Default=None.
+
+        The following sklearn OHE arguments are not supported or are restricted:
+        categories: Not supported since the levels a categorical can assume are those specified by the adaptee CategoricalDimension categories.
+        drop: Not supporting sklearn's OHE drop argument as a array-like of shape (n_features,).
+        dtype: This adapter will always use float64 so we're able to accommodate np.NaN values in the projected/unprojected dataframes.
+        sparse: Not exposed in the adapter since the adapter maintains the OHE instances.
+        handle_unknown: This adapter will always set sklearn's OHE instantiation argument handle_unknown='error'.
+
+    """
+
+    def __init__(
+            self,
+            adaptee: Hypergrid,
+            merge_all_categorical_dimensions: bool = False,
+            drop: str = None
+    ):
+        if not HypergridAdapter.is_like_simple_hypergrid(adaptee):
+            raise ValueError("Adaptee must implement a Hypergrid Interface.")
+
+        HypergridAdapter.__init__(self, name=adaptee.name, random_state=adaptee.random_state)
+
+        self._adaptee: Hypergrid = adaptee
+        self._merge_all_categorical_dimensions = merge_all_categorical_dimensions
+        self._one_hot_encoder_kwargs = {
+            'drop': drop,
+            'dtype': np.float64,
+            'sparse': False,
+            'handle_unknown': 'error'
+        }
+        self._all_one_hot_encoded_target_dimension_names = []
+        self._adaptee_to_target_data_dict = {}

```

Comment:
A type annotation and a comment describing the purpose of this dictionary would help a lot.